### PR TITLE
feat: Support Fieldset to group form fields (#252)

### LINKED ISSUE #252: Support `Fieldset` to group form fields
__Is your feature request related to a problem? Please describe__

Support `Fieldset` to group form fields, counterpart to html tag `fieldset`.

__Describe the solution you'd like__

Add a `Fieldset` field component with type `fieldset`. Suggested schema:
```json
{
      "type": "fieldset",
      "label": "Fieldset",
      "components": [
        {
          "key": "firstName",
          "label": "First Name",
          "type": "textfield",
          "validate": {
            "required": true
          }
        }, 
        {
          "key": "lastName",
          "label": "Last Name",
          "type": "textfield",
          "validate": {
            "required": true
          }
        }
      ]
    }
```

related to #47 

__Describe alternatives you've considered__

__Additional context__

Screenshot_20220511_172022

### PR DESCRIPTION
**Pull request by 9n**
Closes #252

### CODE REVIEW

File: packages/form-js-viewer/test/spec/render/components/form-fields/Fieldset.spec.js
Comment (barmac) [inline packages/form-js-viewer/test/spec/render/components/form-fields/Fieldset.spec.js:9]: To test: Verify that fieldset with children can be rendered.

**Comment by barmac** [comment]
Hi,

Thanks for this PR. It looks good to me. I left one suggestion regarding missing use case. As I am not a form-js expert, let's wait for others before we decide to merge this.

File: packages/form-js-viewer/src/render/components/form-fields/Fieldset.js
Comment (barmac) [inline packages/form-js-viewer/src/render/components/form-fields/Fieldset.js:18]: Legend should not be rendered if label is missing.

**Review by Skaiir — COMMENTED** [review]
One thing I see missing from this is a lack of test coverage of json schema import. We test the playground, the view and the editor import of our .json schema and no .json file has been touched here. 

You can see how this is done here: https://github.com/bpmn-io/form-js/pull/255/.

File: packages/form-js-editor/src/render/components/palette/icons/index.js
Comment (Skaiir) [inline packages/form-js-editor/src/render/components/palette/icons/index.js:3]: Should rename to `FieldsetIcon` to match the naming of the others.

File: packages/form-js-editor/src/render/components/palette/icons/index.js
Comment (Skaiir) [inline packages/form-js-editor/src/render/components/palette/icons/index.js:15]: Renaming to `FieldsetIcon` again, also this list is alphabetically sorted. Little nitpick but we should keep this that way.

File: packages/form-js-viewer/src/render/components/index.js
Comment (Skaiir) [inline packages/form-js-viewer/src/render/components/index.js:16]: List is alphabetically ordered here as well.

**Review by Skaiir — CHANGES_REQUESTED** [review]
To start with, thank you for the amazing contribution, this is a really good starting point and from what I'm currently seeing we will most likely want to integrate this into the codebase after a bit of iteration. 

Please take into consideration all my previous comments, as well as:

#### We need to agree on some styling of fieldsets.

image

The preview of the component looks a bit cramped and sticks out from the rest of our components. We use specific shades of gray for example and we don't usually stick with the base implementation. Also a border radius here would do some good. Feel free to either give it a go yourself or contact me for discussions. I would also be happy to provide a commit for this if you'd like. 

#### Whether we should allow field-sets to be nested or not is quite up to debate. 

While from an HTML standpoint it is valid, some sources cite accessibility concerns ([reference](https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/)). Some others seem not so bothered by this. 

We could also argue that the responsibility lies on the form creator, since html allows this. We may then consider providing warnings for that in the future if we ever implement linting for the form builder.

Inputs from the core team here would be highly appreciated. 

#### I need to align with the core team regarding this contribution against the form development pipeline. 

Right now many changes are in the works for form-js, and fieldsets may be used to enable some of those new features. This is more a concern for the core team but I'm still leaving this in the review as we shouldn't merge this yet until we conduct at least a sync on this.

File: packages/form-js-viewer/src/render/components/form-fields/Fieldset.js
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Fieldset.js:1]: A thing which probably doesn't come from these changes but from previous investigations of nested layout:

The parent is also highlighted when hovering over children. I'd assume only the children are highlighted.

Kapture 2022-05-23 at 13 22 17

Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Fieldset.js:1]: What we also need to consider: is the dashed "children container" still what we want?

image

File: packages/form-js-viewer/src/render/components/index.js
Comment (9n) [reply packages/form-js-viewer/src/render/components/index.js:16]: I don't get you here, `Fieldset` has been alphabetically ordered.

File: packages/form-js-viewer/src/render/components/form-fields/Fieldset.js
Comment (9n) [reply packages/form-js-viewer/src/render/components/form-fields/Fieldset.js:1]: This make me dilemmatic. Dash border looks verbosely, but it can indicates this area may drop-in. If we disallow fieldset nested, dash border may accepted.

File: packages/form-js-viewer/src/render/components/index.js
Comment (Skaiir) [reply packages/form-js-viewer/src/render/components/index.js:16]: I don't get myself either to be honest ❓ 

Sorry about that 20 hours ago me must have missed his coffee 😄

**Comment by Skaiir** [comment]
@9n We're looking at this internally and will keep the PR as a draft for now. 
We'll have an update to work from next week :), thanks for bearing with us.

**Comment by 9n** [comment]
@Skaiir Does this mean you will implements this feature by yourself,  and I should close this pr?

**Comment by Skaiir** [comment]
@9n We can keep the PR open for now, no matter what we're going to do with it in the future it's good to have visibility on it 😄 .

**Comment by Skaiir** [comment]
Closing this one as we're now implementing this quite extensively in the library: https://github.com/bpmn-io/form-js/pull/768

**Pull request closed without merging**
